### ISSUE TITLE
Error using Shopify App Bridge in Storybook: Global object not defined

### ISSUE THREAD

**Issue by masanori-iwata** [Shopify/shopify-app-bridge#325]
## Overview

We're encountering an issue when trying to utilize Shopify's App Bridge within our Storybook environment, leading to errors that hinder the seamless integration of Shopify functionalities into our components. Despite our efforts to configure the Storybook environment to incorporate the App Bridge script, we're faced with an error message indicating that the Shopify global object is not defined. It's worth noting that there might be nuances in our implementation or understanding that could be contributing to this challenge. This situation is particularly significant for us as we aim to leverage Shopify's App Bridge for UI development and testing within Storybook.

#### Packages and Versions

- `@shopify/app-bridge-react` @ "^4.1.2",
- `@shopify/polaris` @ `^12.22.1`,
- `storybook`: `^8.0.4`,
- `vite`: `^5.2.6`,

#### Platform

- OS: macOS
- OS Version: 14.3.1 (23D60)
- Browser and version: Chrome Version 123.0.6312.58 (Official Build) (arm64)
- Storybook version: 8.0

##### File: .storybook/preview-head.html
```
<meta name="shopify-api-key" content="%SHOPIFY_API_KEY%" />
<script src="https://cdn.shopify.com/shopifycloud/app-bridge.js"></script>

```
##### File: .storybook/preview.tsx
```
import React from "react";
import type { Preview } from "@storybook/react";
import "../app/globals.css";
import "@shopify/polaris/build/esm/styles.css";

import { AppProvider } from "@shopify/polaris";
import translations from "@shopify/polaris/locales/en.json";

const preview: Preview = {
  decorators: [
    (StoryFn) => {
      return (
        <AppProvider i18n={translations}>
          <StoryFn />
        </AppProvider>
      );
    },
  ],
  parameters: {
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/i,
      },
    },
  },
};

export default preview;
```

##### File: src/components/CopyButton.tsx
```
import { useAppBridge } from "@shopify/app-bridge-react";
import { Button } from "@shopify/polaris";
import { ClipboardIcon } from "@shopify/polaris-icons";

export const CopyButton = () => {
  const shopify = useAppBridge();
  const copiedMessage = "ID copied.";

  return (
    <Button
      accessibilityLabel={options.ariaLabel}
      icon={ClipboardIcon}
      onClick={() => {
        navigator.clipboard.writeText( ... );
        shopify.toast.show(copiedMessage);
      }}
    />
  );
};

```

#### Error code:
```
Uncaught Error: App Bridge Next: missing required configuration fields: shop
```

```
The shopify global is not defined. This likely means the App Bridge script tag was not added correctly to this page
Error: The shopify global is not defined. This likely means the App Bridge script tag was not added correctly to this page
    at Rr (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/@shopify_app-bridge-react.js?v=33fe01ff:705:11)
    at TimerIDSettingField (http://localhost:6066/app/routes/edit/components/field/basicSettings/TimerIDSettingField.tsx:2:815)
    at renderWithHooks (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:12171:26)
    at mountIndeterminateComponent (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:14921:21)
    at beginWork (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:15902:22)
    at beginWork$1 (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:19749:22)
    at performUnitOfWork (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:19197:20)
    at workLoopSync (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:19133:13)
    at renderRootSync (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:19112:15)
    at recoverFromConcurrentError (http://localhost:6066/node_modules/.cache/storybook/4f7feca90ea19ea5cf329e867b9219ba3dac59ea943b731df82dbe733910037e/sb-vite/deps/chunk-YK4DMFXL.js?v=33fe01ff:18732:28)
```

**Comment by wenlongbrother** [comment]
Hi @masanori-iwata I met this issue also, can you please disclose how you solve it? Thanks

**Comment by masanori-iwata** [comment]
Hi @wenlongbrother,

I haven't solved it yet. However, I saw the link below and thought that App Bridge React might not be intended to work with Storybook. Actually, the modal component is being called from Polaris.

https://storybook.js.org/showcase/shopify-polaris-react
https://github.com/Shopify/polaris/blob/main/polaris-react/src/components/Modal/Modal.stories.tsx

So, I closed this issue because I decided not to use App Bridge React on Storybook.
